### PR TITLE
Update release notes to reflect 3.3.4 release

### PR DESCRIPTION
**Pull request by kovrus**
## Checklist

 - [ ] User relevant changes are recorded in ``CHANGES.txt``
 - [ ] Touched code is covered by tests
 - [ ] Documentation has been updated if necessary
 - [ ] [CLA](https://crate.io/community/contribute/cla/) is signed
 - [ ] This does not contain breaking changes, or if it does:
    - It is released within a major release
    - It is recorded in ``CHANGES.txt``
    - It was marked as deprecated in an earlier release if possible
    - You've thought about the consequences and other components are adapted
      (E.g. AdminUI)

### CODE REVIEW

File: blackbox/docs/appendices/release-notes/unreleased.rst [hunk 1]
```diff
@@ -327,7 +327,8 @@ SQL Standard and PostgreSQL compatibility improvements
   scalar function that yields the current value of the setting.
 
 - Allow :ref:`sql_administration_udf` to be registered against the
-  ``pg_catalog`` schema.
+  ``pg_catalog`` schema. This also extends :ref:`scalar_current_schema` to be
```
Comment (seut) [inline blackbox/docs/appendices/release-notes/unreleased.rst]: This is unrelated to the release, can we at least put it into a dedicated commit?

Comment (kovrus) [reply blackbox/docs/appendices/release-notes/unreleased.rst]: yep, will split the commit

Comment (mfussenegger) [reply blackbox/docs/appendices/release-notes/unreleased.rst]: it wasn't completely unrelated, moved it up from https://github.com/crate/crate/pull/8849/commits/f3aea20b6e801f66948a0e454720bdda5e7b3700#diff-8918b36d12905ff85de79bcbdcf052aaL448

Comment (kovrus) [reply blackbox/docs/appendices/release-notes/unreleased.rst]: hmm shall i revert commit split then?
